azure-pipelines.infra-core.yaml

```diff
@@ -0,0 +1,61 @@
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      name: hmcts/azure-devops-templates
+      ref: refs/heads/master
+      endpoint: hmcts
+
+trigger:
+  - none
+
+parameters:
+- name: environment
+  type: object
+  default:
+   - Dev
+   - Demo
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+variables:
+- name: tfVersion
+  value: 1.1.0
+- name: tfStateServiceConnection
+  value: DTS-SHAREDSERVICES-PROD-Video Hearings
+- group: external
+
+stages:
+
+################################################
+# Check State. #################################
+
+- stage: 'Check_Terraform_State'
+  displayName: Check Terraform State
+  jobs:
+  - job: 'Check_Terraform_State'
+    displayName: Check Terraform State
+    steps:
+    - template: pipeline-steps/check-state.yaml
```

Comment:
RG and Storage Account should be created per environment and then a state file per env